reference/cd-codefresh.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Codefresh
+---
+
+[Codefresh](https://codefresh.io) is a CI/CD platform designed for containers and microservices. It has built-in support for Docker, Kubernetes and Helm.
+
+Codefresh pipelines are composed by different [steps](https://codefresh.io/docs/docs/codefresh-yaml/steps/) where each step runs inside a Docker container.
+Since Pulumi is also released as [Docker container](https://hub.docker.com/r/pulumi/pulumi), it is very easy to use Pulumi in Codefresh pipelines.
+
+
+## Project setup
+
+First of all follow the instructions for creating a Pulumi stack. There are three ways to do this:
+
+1. [Clone an Existing Example](https://github.com/pulumi/examples)
+2. [Use the New Project Wizard](https://app.pulumi.com/site/new-project)
+3. [Download the CLI](https://pulumi.io/quickstart/install.html) and run `pulumi new` to select a template.
+
+Then [signup for a Codefresh account](https://codefresh.io/docs/docs/getting-started/create-a-codefresh-account/) and [create a pipeline](https://codefresh.io/docs/docs/configure-ci-cd-pipeline/pipelines/). There is no special setup needed on the Codefresh side (i.e. you can use Pulumi on both free and paid Codefresh accounts).
+
+
+### Environment Variables
+
+To use Pulumi within Codefresh, there are a few environment variables you'll need to set for each
+build.
+
+The first is `PULUMI_ACCESS_TOKEN`, which is required to authenticate with pulumi.com in order to
+perform the preview or update. You can create a new Pulumi access token specifically for your
+CI/CD job on your [Pulumi Account page](https://app.pulumi.com/account/tokens).
+
+You can either add the token on the pipeline itself as a variable, or store it globally using [Codefresh shared configuration](https://codefresh.io/docs/docs/configure-ci-cd-pipeline/shared-configuration/).
+
+![Pulumi token in Codefresh](/images/reference/codefresh/codefresh-pulumi-token.png)
+
+Next, you will also need to set environment variables specific to your cloud resource provider.
+For example, if your stack is managing resources on AWS, `AWS_ACCESS_KEY_ID` and
+`AWS_SECRET_ACCESS_KEY`.
+
+
+## Codefresh CI/CD pipeline with Pulumi
+
+In your pipeline you need at least two steps.
+
+1. A step that downloads all Pulumi dependencies according to your programming language
+1. A step that performs the actual deployment 
+
+In all cases you use a [Codefresh freestyle step](https://codefresh.io/docs/docs/codefresh-yaml/steps/freestyle/) with the Pulumi Docker image.
+
+```yaml
+  RunMyPulumiStep:
+    title: Running Pulumi inside Codefresh
+    image: pulumi/pulumi
+    commands: 
+      - pulumi stack --non-interactive
```

Comment:
Should this be `pulumi login`?

Reply:
I added it, but the example works without it. I noticed that pulumi logins automatically when a command requires it.